go_retry_poc/main.go

```diff
@@ -0,0 +1,63 @@
+package main
+
+import (
+	"fmt"
+	"math/rand"
+	"time"
+
+	"github.com/cenkalti/backoff/v4"
+)
+
+func main() {
+
+	// * ---------------------------------------------------------------------------------------- *
+	// *                          DEFINE THE ACTION TO BE RETRIED                                 *
+	// * ---------------------------------------------------------------------------------------- *
+
+	// In the case of Aligned, this would be whatever messaging function we would like
+	// to be retried in case of failure.
+	// This function will return two different type of errors: recoverable and not recoverable.
+	// For the case of a recoverable error, the `Retry` function will keep trying until some of the
+	// stop conditions is met.
+	// For the not recoverable case, the `Retry` function will return without retrying again.
+	// This behavior is simulated here with some randomness.
+	// The not recoverable error is simulated by the ``PermanentError`
+	action := func() error {
+		fmt.Println("Doing some operation...")
+		fmt.Printf("Actual time: %v\n", time.Now())
+
+		randomNum := rand.Intn(10)
+		if randomNum > 5 {
+			return backoff.Permanent(fmt.Errorf("Non recoverable error"))
+		}
+
+		return fmt.Errorf("There was an error!")
+	}
+
+	// * ---------------------------------------------------------------------------------------- *
+	// *                         EXPONENTIAL BACKOFF CONFIGURATION                                *
+	// * ---------------------------------------------------------------------------------------- *
+	// For the exponential backoff formula `backoff(n) = a * b^n`, we set the following config:
+	// *    a = 2000ms
+	// * 	b = 2
+	// *    0 <= n <= 2
+	// There is no randomization factor
```

Comment:
We should probably use a single formula, with the constants for that formula, and then derive the parameters for the particular implementation. Something like:
```
Formula: a * b ^ n, with a = 1000ms, b = 2, 1 <= n <= 3
For Rust: n already starts at 1, so we simply pass the values.
For Go: n actually starts at 0 in the library, so we perform the first multiplication on the a parameter so `a * b * b^n-1 = a * b^n`, thus a = 2000ms, b = 2, 0 <= n <= 2.
```
With better phrasing and style than I used, of course. Visually, it should be like the original, but more explicit about the behavior.